crates/server/src/store/config.rs

```diff
@@ -18,6 +18,22 @@ use crate::store::recursor::RecursiveConfig;
 use crate::store::sqlite::SqliteConfig;
 
 /// Enumeration over all Store configurations
+
+/// This is the outer container enum, covering the single- and chained-store variants.
+/// The chained store variant is a vector of StoreConfigs that should be consulted in-order during the lookup process.
+/// An example of this (currently the only example,) is when the blocklist feature is used: the blocklist should be queried first, then
+/// a recursor or forwarder second if the blocklist authority does not match on the query.
+#[derive(Deserialize, PartialEq, Eq, Debug)]
+#[serde(untagged)]
+#[non_exhaustive]
+pub enum StoreConfigContainer {
```

Comment:
Should/could we just simplify this to always having a `Vec`? Presumably `Single(config)` will not be semantically distinguishable from `Chained(vec![config])`?

Reply:
We could, but I don't think it would be a good idea. The enum is just there to make TOML syntax (particularly the terse single-line table/map syntax) more intuitive for end users. If everything was a Vec, then in TOML, all stores would have to be [[stores]], even if there is only one.  With the enum, a user can write [stores] or stores = { ... } for a single store config.

In the server code, there is no semantic difference - the stores are pushed onto a vec whether the user configures one or several, and the users never have to deal with any syntax related to the Enum since it is serde(untagged).